server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/BounceTest.java

```diff
@@ -58,30 +65,104 @@ public void setUp() throws Exception {
         fakeMailContext = FakeMailContext.defaultContext();
 
         when(dnsService.getLocalHost()).thenThrow(new UnknownHostException());
-        bounce.init(new FakeMailetConfig(MAILET_NAME, fakeMailContext));
+    }
+
+    @Test
+    public void getMailetInfoShouldReturnValue() {
+        assertThat(bounce.getMailetInfo()).isEqualTo("Bounce Mailet");
+    }
 
-        senderMailAddress = new MailAddress("sender@domain.com");
-        recipientMailAddress = new MailAddress("recipient@domain.com");
+    @Test
+    public void initShouldThrowWhenUnkownParameter() throws Exception {
+        FakeMailetConfig mailetConfig = new FakeMailetConfig(MAILET_NAME, fakeMailContext);
+        mailetConfig.setProperty("unknown", "error");
+        expectedException.expect(MessagingException.class);
+
+        bounce.init(mailetConfig);
+    }
+
+    @Test
+    public void initShouldNotThrowWhenEveryParameters() throws Exception {
+        FakeMailetConfig mailetConfig = new FakeMailetConfig(MAILET_NAME, fakeMailContext);
+        mailetConfig.setProperty("debug", "true");
+        mailetConfig.setProperty("passThrough", "false");
+        mailetConfig.setProperty("fakeDomainCheck", "false");
+        mailetConfig.setProperty("inline", "all");
+        mailetConfig.setProperty("attachment", "none");
+        mailetConfig.setProperty("message", "custom message");
+        mailetConfig.setProperty("notice", "");
+        mailetConfig.setProperty("sender", "sender@domain.org");
+        mailetConfig.setProperty("sendingAddress", "sender@domain.org");
+        mailetConfig.setProperty("prefix", "my prefix");
+        mailetConfig.setProperty("attachError", "true");
+
+        bounce.init(mailetConfig);
     }
 
     @Test
     public void bounceShouldReturnAMailToTheSenderWithoutAttributes() throws Exception {
+        FakeMailetConfig mailetConfig = new FakeMailetConfig(MAILET_NAME, fakeMailContext);
+        bounce.init(mailetConfig);
+
+        MailAddress senderMailAddress = new MailAddress("sender@domain.com");
         FakeMail mail = FakeMail.builder()
                 .sender(senderMailAddress)
                 .name(MAILET_NAME)
-                .recipient(recipientMailAddress)
+                .recipient(new MailAddress("recipient@domain.com"))
                 .build();
         MimeMessage mimeMessage = new MimeMessage(Session.getDefaultInstance(new Properties()));
         mimeMessage.setText("My content");
         mail.setMessage(mimeMessage);
 
         bounce.service(mail);
 
-        FakeMailContext.SentMail expected = new FakeMailContext.SentMail(null,
-            Lists.newArrayList(senderMailAddress),
-            null,
-            ImmutableMap.<String, Serializable>of());
+        MailAddress sender = null;
+        Collection<MailAddress> recipients = Lists.newArrayList(senderMailAddress);
+        MimeMessage message = null;
```

Comment:
Because we are using FakeMailContext which equals method does not check message equality;
and there's no equals method in java mail MimeMessage class.